### PR TITLE
use compare exchange for the lazy "My" Template

### PR DESCRIPTION
**Pull request by TyOverby**
Closes https://github.com/dotnet/roslyn/issues/17191

### CODE REVIEW

**Comment by TyOverby** [comment]
@AlekseyTs @dotnet/roslyn-compiler

File: src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb [hunk 1]
```diff
@@ -259,13 +260,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                 Throw ExceptionUtilities.Unreachable
                             End If
 
-                            ' set global cache
-                            s_myTemplateCache(parseOptions) = tree
-
-                            _lazyMyTemplate = tree
+                            If Object.Equals(Interlocked.CompareExchange(_lazyMyTemplate, tree, VisualBasicSyntaxTree.Dummy), VisualBasicSyntaxTree.Dummy) Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb]: >Object.Equals [](start = 31, length = 13)

Use ```Is``` operator instead.

**Comment by cston** [comment]
Other than using `Is`, LGTM

File: src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb (hunk 1 above)
Comment (TyOverby) [inline src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb]: Done

**Comment by TyOverby** [comment]
@AlekseyTs: what other embedded trees do we have?

**Comment by TyOverby** [comment]
@AlekseyTs I found all of these: http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Compilation/VisualBasicCompilation.vb,8d9dd0f0f6040de6,references

Looks like they're all threadsafe though

**Pull request merged**